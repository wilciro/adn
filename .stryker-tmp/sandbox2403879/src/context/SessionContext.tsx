// @ts-nocheck
function stryNS_9fa48() {
  var g = new Function("return this")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

import React, { createContext, useEffect, useState } from 'react';
import { SessionState } from 'context/models/SessionState';
import * as sessionService from '../services/sessionService';
export const useStateContainer = (initialState: SessionState = {}) => {
  if (stryMutAct_9fa48("228")) {
    {}
  } else {
    stryCov_9fa48("228");
    const [username, setUsername] = useState(initialState.username);
    const [sessionId, setSessionId] = useState(initialState.sessionId);
    useEffect(() => {
      if (stryMutAct_9fa48("229")) {
        {}
      } else {
        stryCov_9fa48("229");

        if (stryMutAct_9fa48("232") ? false : stryMutAct_9fa48("231") ? true : stryMutAct_9fa48("230") ? username : (stryCov_9fa48("230", "231", "232"), !username)) {
          if (stryMutAct_9fa48("233")) {
            {}
          } else {
            stryCov_9fa48("233");
            setSessionId(sessionService.deleteSession());
            return;
          }
        }

        setSessionId(sessionService.createSession(username));
      }
    }, stryMutAct_9fa48("234") ? [] : (stryCov_9fa48("234"), [username]));
    return stryMutAct_9fa48("235") ? {} : (stryCov_9fa48("235"), {
      data: stryMutAct_9fa48("236") ? {} : (stryCov_9fa48("236"), {
        username,
        sessionId
      }),
      mutations: stryMutAct_9fa48("237") ? {} : (stryCov_9fa48("237"), {
        setUsername
      })
    });
  }
};
export const SessionContext = createContext<ReturnType<typeof useStateContainer>>(({} as never));
interface SessionProps {
  children: React.ReactNode;
}
export const SessionProvider = ({
  children
}: SessionProps) => {
  if (stryMutAct_9fa48("238")) {
    {}
  } else {
    stryCov_9fa48("238");
    const contextValue = useStateContainer(sessionService.recoverSession());
    return <SessionContext.Provider value={contextValue}>
      {children}
    </SessionContext.Provider>;
  }
};