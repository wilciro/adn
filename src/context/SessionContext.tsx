import React, { createContext, useEffect, useRef, useState } from 'react';
import { SessionState } from 'context/models/SessionState';
import * as sessionService from '../services/sessionService';

export const useStateContainer = (initialState: SessionState = {}) => {
  const [username, setUsername] = useState(initialState.username);
  const [sessionId, setSessionId] = useState(initialState.sessionId);

  useEffect(() => {
    if (!username) {
      setSessionId(sessionService.deleteSession());
      return;
    }
    setSessionId(sessionService.createSession(username));
  }, [username]);

  return {
    data: { username, sessionId },
    mutations: { setUsername },
  };
};

export const SessionContext = createContext<
  ReturnType<typeof useStateContainer>
>({} as never);

interface SessionProps {
  children: React.ReactNode;
}

export const SessionProvider = ({ children }: SessionProps) => {
  const contextValue = useStateContainer(sessionService.recoverSession());
  return (
    <SessionContext.Provider value={contextValue}>
      {children}
    </SessionContext.Provider>
  );
};

