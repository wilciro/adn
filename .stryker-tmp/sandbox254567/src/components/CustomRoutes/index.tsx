// @ts-nocheck
// React
function stryNS_9fa48() {
  var g = new Function("return this")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

import React, { FC, Suspense, useContext } from 'react';
import { Route } from 'react-router-dom';
import { Navigate, Routes } from 'react-router'; // Context

import { SessionContext } from 'context/SessionContext'; // Pages

import LandingPage from 'pages/Landing';
import RequestsListPage from 'pages/RequestsList';
import AboutPage from 'pages/About';
const LoginPage = React.lazy(stryMutAct_9fa48("45") ? () => undefined : (stryCov_9fa48("45"), () => import(stryMutAct_9fa48("46") ? "" : (stryCov_9fa48("46"), 'pages/Login'))));

const CustomRoutes: FC = () => {
  if (stryMutAct_9fa48("47")) {
    {}
  } else {
    stryCov_9fa48("47");
    const {
      data: {
        username
      }
    } = useContext(SessionContext);
    const logged = stryMutAct_9fa48("48") ? username && false : (stryCov_9fa48("48"), username ?? (stryMutAct_9fa48("49") ? true : (stryCov_9fa48("49"), false)));
    return <Routes>
      <Route path="/404" element={<LandingPage />} />
      <Route path="/login" element={(stryMutAct_9fa48("50") ? logged : (stryCov_9fa48("50"), !logged)) ? <Suspense fallback={<div>Loading...</div>}>
              <LoginPage />
            </Suspense> : <Navigate to="/dashboard" />} />
      <Route path="/dashboard" element={logged ? <RequestsListPage /> : <Navigate to="/login" />} />
      <Route path="/" element={<LandingPage />} />
      <Route path="/about" element={<AboutPage />} />
      <Route path="*" element={<Navigate to="/404" replace />} />
    </Routes>;
  }
};

export default CustomRoutes;