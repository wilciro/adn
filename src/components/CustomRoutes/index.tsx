// React
import React, { FC, Suspense, useContext } from 'react';
import { Route } from 'react-router-dom';
import { Navigate, Routes } from 'react-router';

// Context
import { SessionContext } from 'context/SessionContext';

// Pages
import LandingPage from 'pages/Landing';
import RequestsListPage from 'pages/RequestsList';
import AboutPage from 'pages/About';

const LoginPage = React.lazy(() => import('pages/Login'));

const CustomRoutes: FC = () => {

  const { data: {username} } = useContext(SessionContext);
  const logged = username ?? false;

  return (
    <Routes>      
        <Route
            path="/404"
            element={<LandingPage />}
        />
        <Route
            path="/login"
            element={ 
                !logged ?
                    <Suspense fallback={<div>Loading...</div>}>
                        <LoginPage /> 
                    </Suspense>
                    : 
                    <Navigate to="/dashboard" /> }
        />
        <Route
            path="/dashboard"
            element={ logged ?  <RequestsListPage /> : <Navigate to="/login" /> }
        />
        <Route
            path="/"
            element={<LandingPage />}
        />
        <Route
            path="/about"
            element={<AboutPage />}
        />
        <Route path="*" element={<Navigate to="/404" replace />} />
    </Routes>
  );
};

export default CustomRoutes;
