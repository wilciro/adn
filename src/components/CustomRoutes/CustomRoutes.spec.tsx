import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SessionContext, SessionProvider } from 'context/SessionContext';
import { Button } from '@mantine/core';
import CustomRoutes from '.';

describe('CustomRoutes tests', () => {
  const TestComponent = () => {
    const { mutations } = React.useContext(SessionContext);
    return (
      <MemoryRouter initialEntries={['/dashboard']}>
        <CustomRoutes />
        <Button
          data-testid="login"
          onClick={() => {
            mutations.setUsername('Juan');
          }}
        >
          login
        </Button>
        <Button
          data-testid="logout"
          onClick={() => {
            mutations.setUsername(undefined);
          }}
        >
          logout
        </Button>
      </MemoryRouter>
    );
  };

  it('should go to login', () => {
    const { container } = render(
      <SessionProvider>
        <TestComponent />
      </SessionProvider>,
    );
    expect(screen.getAllByTestId('title-login')).toHaveLength(1);
  });
  it('should match login routes', () => {
    waitFor(() => {
      const { container } = render(
        <SessionProvider>
          <TestComponent />
        </SessionProvider>,
      );
    });
    fireEvent.click(screen.getByTestId('login'));
    waitFor(() => {
      expect(screen.getAllByTestId('header-landing')).toHaveLength(1);
    });
  });
});
