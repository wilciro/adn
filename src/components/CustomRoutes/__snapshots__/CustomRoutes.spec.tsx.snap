// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomRoutes tests should match login routes 1`] = `
<div>
  <div
    class="mantine-yv0or0"
  >
    <section
      class="sc-bczRLJ kuDUnz"
    >
      <h2>
        Total: 
        $ 0,00
      </h2>
    </section>
    <h2>
      Solicitud de cita veterinaria
    </h2>
    <p>
      Diligencia todos los campos para asignar la cita
    </p>
    <div
      class="mantine-1ne3752"
    />
    <form>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-mwamasyp3"
            id="mantine-mwamasyp3-label"
          >
            Nombre del dueño
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="ownerName"
              id="mantine-mwamasyp3"
              placeholder="Ingrese el nombre del dueño"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-ek8kiup46"
            id="mantine-ek8kiup46-label"
          >
            Documento del dueño
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="ownerDocument"
              id="mantine-ek8kiup46"
              placeholder="Ingrese el documento del dueño"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-t8wyonscb"
            id="mantine-t8wyonscb-label"
          >
            Nombre de la mascota
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="petName"
              id="mantine-t8wyonscb"
              placeholder="Ingrese el nombre de la mascota"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-NumberInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-NumberInput-label"
            for="mantine-iyhd1g5ct"
            id="mantine-iyhd1g5ct-label"
          >
            Edad de la mascota
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-NumberInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-NumberInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-NumberInput-defaultVariant mantine-NumberInput-input mantine-115jdka"
              data-testid="petAge"
              id="mantine-iyhd1g5ct"
              inputmode="numeric"
              placeholder="Ingrese la edad de la mascota"
              required=""
              step="1"
              type="text"
              value="0"
            />
            <div
              class="mantine-1cwybh2 mantine-NumberInput-rightSection"
            >
              <div
                class="mantine-p0axji mantine-NumberInput-rightSection"
              >
                <button
                  aria-hidden="true"
                  class="mantine-NumberInput-control mantine-NumberInput-controlUp mantine-jbog3w"
                  tabindex="-1"
                  type="button"
                />
                <button
                  aria-hidden="true"
                  class="mantine-NumberInput-control mantine-NumberInput-controlDown mantine-xb3ria"
                  tabindex="-1"
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-Select-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-Select-label"
            for="mantine-2fuuai2yk"
            id="mantine-2fuuai2yk-label"
          >
            Tipo de mascota
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-Select-required"
            >
               *
            </span>
          </label>
          <div
            aria-controls="mantine-2fuuai2yk"
            aria-expanded="false"
            aria-haspopup="listbox"
            role="combobox"
            tabindex="-1"
          >
            <input
              type="hidden"
              value=""
            />
            <div
              class="mantine-Select-wrapper mantine-12sbrde"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocomplete="nope"
                class="mantine-Select-defaultVariant mantine-Select-input mantine-1kkmtua"
                data-mantine-stop-propagation="false"
                data-testid="petType"
                id="mantine-2fuuai2yk"
                placeholder="Seleccione el tipo de mascota"
                readonly=""
                required=""
                type="text"
                value=""
              />
              <div
                class="mantine-uquezq mantine-Select-rightSection"
              >
                <svg
                  data-chevron="true"
                  fill="none"
                  height="18"
                  style="color: rgb(134, 142, 150);"
                  viewBox="0 0 15 15"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-DatePicker-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-DatePicker-label"
            for="mantine-4gwmdyuy6"
            id="mantine-4gwmdyuy6-label"
          >
            Fecha de la cita
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-DatePicker-required"
            >
               *
            </span>
          </label>
          <div>
            <div
              class="mantine-2uofp0 mantine-DatePicker-wrapper"
            >
              <div
                class="mantine-DatePicker-wrapper mantine-12sbrde"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="mantine-DatePicker-defaultVariant mantine-DatePicker-input mantine-DatePicker-input mantine-1oqua4f"
                  data-testid="date"
                  id="mantine-4gwmdyuy6"
                  name="date"
                  placeholder="Seleccione la fecha de la cita"
                  readonly=""
                  required=""
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TimeInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TimeInput-label"
            for="mantine-nssua1cio"
            id="mantine-nssua1cio-label"
          >
            Hora de la cita
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TimeInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TimeInput-wrapper mantine-12sbrde"
          >
            <div
              aria-invalid="false"
              class="mantine-TimeInput-defaultVariant mantine-TimeInput-input mantine-g773hu"
              data-testid="time"
              placeholder="Seleccione la hora de la cita"
              required=""
            >
              <div
                class="mantine-x78oh3 mantine-TimeInput-controls"
              >
                <input
                  class="mantine-TimeInput-timeInput mantine-ix2kpu"
                  id="mantine-nssua1cio"
                  inputmode="numeric"
                  placeholder="--"
                  type="text"
                  value="11"
                />
                <div
                  class="mantine-Text-root mantine-2qa0ve"
                  style="line-height: 1;"
                >
                  :
                </div>
                <input
                  class="mantine-TimeInput-timeInput mantine-ix2kpu"
                  inputmode="numeric"
                  placeholder="--"
                  type="text"
                  value="23"
                />
                <input
                  class="mantine-kolqo2"
                  placeholder="am"
                  type="text"
                  value="am"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
    <div
      class="mantine-1ne3752"
    />
    <button
      class="mantine-Button-filled mantine-Button-root mantine-1grzg0q"
      data-testid="register-btn"
      type="button"
    >
      <div
        class="mantine-3xbgk5 mantine-Button-inner"
      >
        <span
          class="mantine-qo1k2 mantine-Button-label"
        >
          Registrar
        </span>
      </div>
    </button>
  </div>
</div>
`;

exports[`CustomRoutes tests should match snapshot CustomRoutes 1`] = `
<div>
  <div
    class="mantine-yv0or0"
  >
    <section
      class="sc-bczRLJ kuDUnz"
    >
      <h2>
        Total: 
        $ 0,00
      </h2>
    </section>
    <h2>
      Solicitud de cita veterinaria
    </h2>
    <p>
      Diligencia todos los campos para asignar la cita
    </p>
    <div
      class="mantine-1ne3752"
    />
    <form>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-bpalbvkql"
            id="mantine-bpalbvkql-label"
          >
            Nombre del dueño
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="ownerName"
              id="mantine-bpalbvkql"
              placeholder="Ingrese el nombre del dueño"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-va89t69gx"
            id="mantine-va89t69gx-label"
          >
            Documento del dueño
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="ownerDocument"
              id="mantine-va89t69gx"
              placeholder="Ingrese el documento del dueño"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TextInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TextInput-label"
            for="mantine-4lita9btj"
            id="mantine-4lita9btj-label"
          >
            Nombre de la mascota
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TextInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TextInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-TextInput-defaultVariant mantine-TextInput-input mantine-g773hu"
              data-testid="petName"
              id="mantine-4lita9btj"
              placeholder="Ingrese el nombre de la mascota"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-NumberInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-NumberInput-label"
            for="mantine-qfqd10ij0"
            id="mantine-qfqd10ij0-label"
          >
            Edad de la mascota
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-NumberInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-NumberInput-wrapper mantine-12sbrde"
          >
            <input
              aria-invalid="false"
              class="mantine-NumberInput-defaultVariant mantine-NumberInput-input mantine-115jdka"
              data-testid="petAge"
              id="mantine-qfqd10ij0"
              inputmode="numeric"
              placeholder="Ingrese la edad de la mascota"
              required=""
              step="1"
              type="text"
              value="0"
            />
            <div
              class="mantine-1cwybh2 mantine-NumberInput-rightSection"
            >
              <div
                class="mantine-p0axji mantine-NumberInput-rightSection"
              >
                <button
                  aria-hidden="true"
                  class="mantine-NumberInput-control mantine-NumberInput-controlUp mantine-jbog3w"
                  tabindex="-1"
                  type="button"
                />
                <button
                  aria-hidden="true"
                  class="mantine-NumberInput-control mantine-NumberInput-controlDown mantine-xb3ria"
                  tabindex="-1"
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-Select-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-Select-label"
            for="mantine-2hbyhkas3"
            id="mantine-2hbyhkas3-label"
          >
            Tipo de mascota
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-Select-required"
            >
               *
            </span>
          </label>
          <div
            aria-controls="mantine-2hbyhkas3"
            aria-expanded="false"
            aria-haspopup="listbox"
            role="combobox"
            tabindex="-1"
          >
            <input
              type="hidden"
              value=""
            />
            <div
              class="mantine-Select-wrapper mantine-12sbrde"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocomplete="nope"
                class="mantine-Select-defaultVariant mantine-Select-input mantine-1kkmtua"
                data-mantine-stop-propagation="false"
                data-testid="petType"
                id="mantine-2hbyhkas3"
                placeholder="Seleccione el tipo de mascota"
                readonly=""
                required=""
                type="text"
                value=""
              />
              <div
                class="mantine-uquezq mantine-Select-rightSection"
              >
                <svg
                  data-chevron="true"
                  fill="none"
                  height="18"
                  style="color: rgb(134, 142, 150);"
                  viewBox="0 0 15 15"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.93179 5.43179C4.75605 5.60753 4.75605 5.89245 4.93179 6.06819C5.10753 6.24392 5.39245 6.24392 5.56819 6.06819L7.49999 4.13638L9.43179 6.06819C9.60753 6.24392 9.89245 6.24392 10.0682 6.06819C10.2439 5.89245 10.2439 5.60753 10.0682 5.43179L7.81819 3.18179C7.73379 3.0974 7.61933 3.04999 7.49999 3.04999C7.38064 3.04999 7.26618 3.0974 7.18179 3.18179L4.93179 5.43179ZM10.0682 9.56819C10.2439 9.39245 10.2439 9.10753 10.0682 8.93179C9.89245 8.75606 9.60753 8.75606 9.43179 8.93179L7.49999 10.8636L5.56819 8.93179C5.39245 8.75606 5.10753 8.75606 4.93179 8.93179C4.75605 9.10753 4.75605 9.39245 4.93179 9.56819L7.18179 11.8182C7.35753 11.9939 7.64245 11.9939 7.81819 11.8182L10.0682 9.56819Z"
                    fill="currentColor"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-DatePicker-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-DatePicker-label"
            for="mantine-a43xgexqp"
            id="mantine-a43xgexqp-label"
          >
            Fecha de la cita
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-DatePicker-required"
            >
               *
            </span>
          </label>
          <div>
            <div
              class="mantine-2uofp0 mantine-DatePicker-wrapper"
            >
              <div
                class="mantine-DatePicker-wrapper mantine-12sbrde"
              >
                <input
                  aria-invalid="false"
                  autocomplete="off"
                  class="mantine-DatePicker-defaultVariant mantine-DatePicker-input mantine-DatePicker-input mantine-1oqua4f"
                  data-testid="date"
                  id="mantine-a43xgexqp"
                  name="date"
                  placeholder="Seleccione la fecha de la cita"
                  readonly=""
                  required=""
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="mantine-TimeInput-root mantine-1m3pqry"
        >
          <label
            class="mantine-1w07r5r mantine-TimeInput-label"
            for="mantine-vy4a4h31r"
            id="mantine-vy4a4h31r-label"
          >
            Hora de la cita
            <span
              aria-hidden="true"
              class="mantine-1m203yh mantine-TimeInput-required"
            >
               *
            </span>
          </label>
          <div
            class="mantine-TimeInput-wrapper mantine-12sbrde"
          >
            <div
              aria-invalid="false"
              class="mantine-TimeInput-defaultVariant mantine-TimeInput-input mantine-g773hu"
              data-testid="time"
              placeholder="Seleccione la hora de la cita"
              required=""
            >
              <div
                class="mantine-x78oh3 mantine-TimeInput-controls"
              >
                <input
                  class="mantine-TimeInput-timeInput mantine-ix2kpu"
                  id="mantine-vy4a4h31r"
                  inputmode="numeric"
                  placeholder="--"
                  type="text"
                  value="11"
                />
                <div
                  class="mantine-Text-root mantine-2qa0ve"
                  style="line-height: 1;"
                >
                  :
                </div>
                <input
                  class="mantine-TimeInput-timeInput mantine-ix2kpu"
                  inputmode="numeric"
                  placeholder="--"
                  type="text"
                  value="23"
                />
                <input
                  class="mantine-kolqo2"
                  placeholder="am"
                  type="text"
                  value="am"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
    <div
      class="mantine-1ne3752"
    />
    <button
      class="mantine-Button-filled mantine-Button-root mantine-1grzg0q"
      data-testid="register-btn"
      type="button"
    >
      <div
        class="mantine-3xbgk5 mantine-Button-inner"
      >
        <span
          class="mantine-qo1k2 mantine-Button-label"
        >
          Registrar
        </span>
      </div>
    </button>
  </div>
</div>
`;
